### PR TITLE
Fix up field types

### PR DESCRIPTION
**Pull request by joshgachnang**
### **User description**
Make it so Field understands what props you can pass based on type.

___

### **PR Type**
Enhancement, Bug fix

___

### **Description**
- Simplified `AddressField` component props and default values.
- Refactored field props interfaces in `Common.ts` for better type safety.
- Added `FC` type to `Field` component and adjusted type handling.
- Updated `TapToEdit` component and commented out unsupported field types.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>AddressField.tsx</strong><dd><code>Simplify AddressField component props and default values</code>&nbsp; </dd></summary>
<hr>

packages/ui/src/AddressField.tsx

<li>Removed <code>Partial<FieldProps></code> from <code>AddressField</code> component props.<br> <li> Simplified default value assignment for <code>AddressInterface</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/629/files#diff-4b4ce63ea8410caaeb7b2926e02e8241c57fbb3aa1dd9f591ef0473c302d189d">+3/-4</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Common.ts</strong><dd><code>Refactor field props interfaces for better type safety</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

packages/ui/src/Common.ts

<li>Removed redundant <code>FieldProps</code> interface.<br> <li> Updated <code>AddressFieldProps</code> to include <code>value</code>, <code>onChange</code>, and <code>onBlur</code>.<br> <li> Modified <code>TextAreaProps</code> to use <code>Omit</code> instead of <code>Exclude</code>.<br> <li> Refactored <code>FieldProps</code> to be a union of specific field props.<br>

</details>

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/629/files#diff-538e93b553971318d69ff6fff2ac103f295c4922e24bbe61c30897baa463bba1">+32/-55</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>Field.tsx</strong><dd><code>Add FC type and adjust type handling in Field component</code>&nbsp; &nbsp; </dd></summary>
<hr>

packages/ui/src/Field.tsx

<li>Added <code>FC</code> type to <code>Field</code> component.<br> <li> Adjusted handling of <code>type</code> prop for <code>TextField</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/629/files#diff-b179881630547b40643ddd11ccdc12624a62bf6deafcea924246db78416d25d5">+11/-10</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>TapToEdit.tsx</strong><dd><code>Update TapToEdit component and comment out unsupported field types</code></dd></summary>
<hr>

packages/ui/src/TapToEdit.tsx

<li>Removed <code>placeholder</code> prop from <code>TapToEdit</code> component.<br> <li> Adjusted <code>Field</code> component usage within <code>TapToEdit</code>.<br> <li> Commented out handling for <code>percent</code> and <code>currency</code> field types.<br>

</details>

  </td>
  <td><a href="https://github.com/FlourishHealth/ferns-ui/pull/629/files#diff-5c106f38db848acb6a5d382743af81c919163754380b895781575765bc292ecd">+16/-17</a>&nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: packages/ui/src/TapToEdit.tsx [hunk 1]
```diff
@@ -122,10 +121,10 @@ export const TapToEdit = ({
           <Field
             helperText={description}
             label={title}
-            placeholder={placeholder}
+            type={(fieldProps?.type ?? "text") as NonNullable<FieldProps["type"]>}
```
Comment (swanka) [inline packages/ui/src/TapToEdit.tsx:124]: Did not know this was a thing!

File: packages/ui/src/Common.ts [hunk 2]
```diff
@@ -1529,44 +1524,26 @@ export interface ErrorPageProps {
...
-  googleMapsApiKey?: string;
-  googlePlacesMobileStyles?: Styles;
-  transformValue?: TransformValueOptions;
-}
+export type FieldProps =
+  | TextFieldProps
+  | NumberFieldProps
+  | NumberRangeFieldProps
+  | DateTimeFieldProps
+  | (MultiselectFieldProps & {type: "multiselect"})
```
Comment (swanka) [inline packages/ui/src/Common.ts:1532]: Can this type not be enforced in the interface definition?

**Review by swanka — APPROVED** [review]
Just one question. Otherwise looks great!

File: packages/ui/src/TapToEdit.tsx (hunk 1 above)
Comment (joshgachnang) [reply packages/ui/src/TapToEdit.tsx:124]: It's very rare I've run into it, but super helpful here! The real fix is probably forcing type to always be provided, but I couldn't convince the interface to do that.

File: packages/ui/src/Common.ts (hunk 2 above)
Comment (joshgachnang) [reply packages/ui/src/Common.ts:1532]: Do you mean making type="multiselect" on the interface? I went this route so you could do <MultiSelectField ... /> and not have to provide type="multiselect" each time, but you would have to provide it to <Field /> so it knows what to do.

Comment (swanka) [reply packages/ui/src/Common.ts:1532]: Ahhh that makes sense

**Pull request merged**